File: src/main/java/algoandusecases/listscenariosjava/SumOfDigits.java
package algoandusecases.listscenariosjava;

public class SumOfDigits {
    public static void main(String[] args) {
        int digit = 9154;
        System.out.println(getSumOfDigits(digit));

    }
    public static int getSumOfDigits(int number)
    {
        int sum = 0;
        if(number < 10)
        {
            return number;
        }
        while(number>0)
        {
            int remainder =number % 10;
            sum = sum + remainder;
            number = (int)number/10;
        }
        return sum;
    }
}

File: src/main/scala/algoandusecases/listscenarios/SumOfDigits.scala
package algoandusecases.listscenarios

object SumOfDigits {
  def main(args: Array[String]): Unit = {
    val number = 1234;
    //println(sumOfDigits(number))
    println(sumOfDigitsRec(number,0))

  }
  def sumOfDigits(number:Integer):Integer=
  {
    var sum=0
    var n = number
    while(n>0)
      {
        sum = sum + n%10
        n=n/10
      }
      return sum

  }
  def sumOfDigitsRec(number:Integer,sum:Integer):Integer=
  {
    var localSum = sum
    var localNumber = number

    if(localNumber > 0)
      {
        localSum = localSum + localNumber%10
        localNumber=localNumber/10
        return sumOfDigitsRec(localNumber,localSum)
      }
    else
      return localSum

  }


}
